main.rs

```diff
@@ -7,68 +7,36 @@
 #![deny(unused_must_use)]
 #![deny(elided_lifetimes_in_paths)]
 
-use std::{path::PathBuf, str::FromStr};
-
 use clap::Parser;
 use logger::initialise_logging_global;
-use resource::constants::server::{ASCII_LOGO, DISTRIBUTION, SENTRY_REPORTING_URI, VERSION};
-use sentry::ClientInitGuard as SentryGuard;
-use server::parameters::{
-    cli::CLIArgs,
-    config::{Config, DiagnosticsConfig, EncryptionConfig, ServerConfig},
-};
+use resource::constants::server::{DISTRIBUTION, SENTRY_REPORTING_URI, VERSION};
+use server::parameters::config::Config;
+use server::parameters::cli::CLIArgs;
+use server::server::Server;
+use tokio::runtime::Runtime;
 
 fn main() {
-    setup_abort_on_panic();
+    let config = CLIArgs::parse().to_config();
```

Comment:
I thought you were going to move all this to Server?

Reply:
the config object required by the `may_initialise_error_reporting` on the `main` function unfortunately!

Reply:
you could move that into server too though?

Reply:
Works for me this way too tho, approved